File: composeApp/src/androidMain/kotlin/fr/thomasbernard03/tarot/commons/extensions/PlayerColorExtensions.kt
package fr.thomasbernard03.tarot.commons.extensions

import androidx.compose.ui.graphics.Color
import domain.models.PlayerColor
import fr.thomasbernard03.tarot.presentation.theme.Blue
import fr.thomasbernard03.tarot.presentation.theme.Brown
import fr.thomasbernard03.tarot.presentation.theme.Green
import fr.thomasbernard03.tarot.presentation.theme.Orange
import fr.thomasbernard03.tarot.presentation.theme.Pink
import fr.thomasbernard03.tarot.presentation.theme.Purple
import fr.thomasbernard03.tarot.presentation.theme.Red
import fr.thomasbernard03.tarot.presentation.theme.Yellow

fun PlayerColor.toColor() : Color {
    return when(this) {
        PlayerColor.RED -> Red
        PlayerColor.BLUE -> Blue
        PlayerColor.GREEN -> Green
        PlayerColor.ORANGE -> Orange
        PlayerColor.PURPLE -> Purple
        PlayerColor.YELLOW -> Yellow
        PlayerColor.PINK -> Pink
        PlayerColor.BROWN -> Brown
    }
}

File: iosApp/iosApp/Commons/Extensions/PlayerColorExtensions.swift
//
//  PlayerColorExtensions.swift
//  iosApp
//
//  Created by Thomas Bernard on 15/07/2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import Shared
import SwiftUI

extension PlayerColor {
    func toColor() -> Color {
        switch self {
        case .red:
            return Color.red
        case .blue:
            return Color.blue
        case .green:
            return Color.green
        case .orange:
            return Color.orange
        case .purple:
            return Color.purple
        case .yellow:
            return Color.yellow
        case .pink:
            return Color.pink
        case .brown:
            return Color.brown
        default:
            return Color.black
        }
    }
    
    static func all() -> [PlayerColor] {
        return [.red, .blue, .green, .orange, .purple, .yellow, .pink, .brown]
    }
}
